전남대 FE_강예람_1주차 과제 Step 2,3

yeramm opened the pull request:
안녕하세요. FE 1주차 과제 Step2,3 올립니다.

정말 죄송한 말씀이지만, 사실 Step2의 대부분을 수행하지 못했습니다.

![에러](https://github.com/kakao-tech-campus-2nd-step2/react-gift-react-foundation/assets/105156288/3cf7ba78-7917-4bea-9f96-35d8f81b06c4)
![에러2](https://github.com/kakao-tech-campus-2nd-step2/react-gift-react-foundation/assets/105156288/09986db6-63b9-4fa0-8f6a-cef460471983)

storybook을 설치한 후 실행하니 위와 같은 오류가 발생했고,
이를 해결하기 위해 여러 자료를 살펴보았지만 해결하지 못한 상황입니다.

컴파일 자체가 안되니 이후 과제도 진행하지 못했고 그리하여 미완료된 과제를 올리게되었습니다.
낮 시간에 미리 코치님께 여쭤보았어야 했는데 그러지 못해 정말 죄송합니다.

yeramm committed:
docs: READ.md 수정

READ.md에 구현할 기능 목록 추가

yeramm committed:
chore: TypeScript로 동작되는 React 프로젝트 생성

Create React App 기반으로 프로젝트를 생성
Typescript로 프로젝트가 동작됨

yeramm committed:
chore: 절대 경로 import를 위한 alias 설정

절대 경로로 import 할 수 있게 alias 를 설정함

yeramm committed:
chore: 필요없는 파일 및 코드 삭제

필요하지 않은 파일 및 코드 삭제

yeramm committed:
chore: ESLint 추가 및 Airbnb룰 설정

ESLint를 추가하고 Airbnb룰을 설정함

yeramm committed:
chore: Prettier 추가

Prettier를 추가함

yeramm committed:
chore: emotion 라이브러리 추가 및 reset css 적용

emotion 라이브러리를 추가하고 reset css를 적용함

yeramm committed:
docs: READ.md에 Step2 요구사항 추가

READ.md에 Step2 요구사항을 추가하여 정리함

yeramm committed:
docs: READ.md 수정

Step1 요구사항임을 명시하기위해 수정함

yeramm committed:
docs: READ.md에 step2 요구사항 추가

READ.md에 step2 요구사항을 정리하여 추가함

yeramm committed:
chore: storybook 설정

storybook을 설정했으나
eslint, prettier 충돌로 인해 컴파일 오류 발생

yeramm committed:
docs: README.md에 개발 애로사항 추가

README.md에 개발 중  애로사항을 추가함

yeramm committed:
docs: 1주차 질문 답변

READ.md에 1주차 질문 답변 추가

wacilpong commented:
에러는 eslint-config-prettier가 없거나 문제가 있어서 나는 에러로 보이는데요. 아래대로 의존성을 추가하고 eslint 설정파일을 수정했을 때는 발생하나요?

```
npm install --save-dev eslint-config-prettier
```
```
// .eslintrc.json
...
extends: [
  ...
  "eslint-config-prettier"
]
...

yeramm commented:
네.. 의존성을 추가해도 같은 오류가 발생합니다 ㅠㅠ

yeramm committed:
fix: storybook 실행시 발생했던 컴파일 오류 수정

storybook 실행시 발생했던 eslint-config-prettier 관련 오류 해결했으나
prettier 관련 오류 발생

yeramm commented:
1. package.json 파일 수정
2. npm remove eslint @typescript-eslint/parser @typescript-eslint/eslint-plugin 으로 eslint 관련 모든 파일 삭제
3. npx eslint --init으로  eslint 재설치
위 과정을 하니 기존 오류는 발생되지 않았습니다만

아래 오류가 새롭게 떠서.. 해결 중입니다..
![image](https://github.com/kakao-tech-campus-2nd-step2/react-gift-react-foundation/assets/105156288/757ef200-3954-4419-9b71-b09fe78b83ba)
![image](https://github.com/kakao-tech-campus-2nd-step2/react-gift-react-foundation/assets/105156288/a6ca036c-04e2-44f8-b450-fd86290e0dba)

yeramm committed:
fix: eslint와 prettier 간 버전 호환성 해결

eslint, eslint-plugin-prettier, eslint-config-prettier을 재설치하여
버전 호환성을 해결하고자 함
@typescript-eslint/parser 패키지도 설치

yeramm committed:
fix: Quotes 및 Prettier 관련 오류 해결

.prettierrc 파일 및 .eslintrc.json 파일을 수정하여
이중 인용부호와 줄바꿈 문자 스타일 관련 문제를 해결함

wacilpong commented:
@yeramm 
예람님~ 본래 프로젝트 설정하는 부분이 가장 어려운 것이 사실입니다!
그래서 실제 스펙을 개발하는 것과 달리 설정부분은 best practice와 정답이 반드시 존재하는 부분이니만큼,
오픈소스 코드나 다른 멘티분들의 소스코드를 참고해서 확인해보는 것도 좋은 답안일 수 있습니다.

설정하는 작업에서 소위 삽질을 너무 오래 하는 것은 신입레벨에서는 시간사용에 있어서 조금 아쉬울 수 있습니다.
시간은 한정적이고 배워야할 것은 너무 많으니까요~
한정적인 시간 내에서 해결해야 한다면, 스토리북이 설정된 오픈소스나 다른 멘티분들 PR을 참고해보시면 어떨까요?

yeramm commented:
넵 감사합니다 ㅎㅎ 오픈소스나 다른 PR 많이 참고해보겠습니다!

wacilpong commented:
@yeramm 
넵넵! 하지만 문제를 파고드는 부분은 개발자에게 있어서 정말 중요한 부분이라
잘 하고 있다고 생각합니다 👍👍

다만 과제진행에 있어서 말씀드린 것이니 참고해주세요 ㅎㅎ
